### DISCUSSION TITLE
Run Task Error - Operation with dblink

### DISCUSSION THREAD

**Discussion by albercik777** [dbeaver/dbeaver#18234] [General]
Hello,

I have some issu with DBeaver and running Task with Sql Script - updating table at DBLinked Table.

I have MS SQL Server (2016) - let's call it: Srv1, there is an DB Link to other MS SQL Server - let's call it: Srv2.
If I run query to table at Srv2, from Srv1 - it works fine
`select *  from DbLinkName.Database.Scheme.TableName;`
`delete from DbLinkName.Database.Scheme.TableName;`

but when i save it in a Script.sql and try to run by task, i got error:

> 2022-11-16 15:04:03.255 - SQL Scripts Execute
> 2022-11-16 15:04:03.261 - > Execute script [/General/Scripts/TASKS/Script.sql] in [Srv1]
> 2022-11-16 15:04:03.261 - > Default catalog: XXX
> 2022-11-16 15:04:03.261 - > Default schema: xxx
> 2022-11-16 15:04:03.262 - 	Set JDBC connection auto-commit false
> 2022-11-16 15:04:03.263 - Execute queries (1)
> 2022-11-16 15:04:03.264 - -----------------> Execute query
> {Query command}
> 
> 2022-11-16 15:04:03.372 - -----------------> Time: 0ms
> 2022-11-16 15:04:03.373 - org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [7391] [S0002]: Nie można wykonać operacji, ponieważ dostawca OLE DB „SQLNCLI11” dla połączonego serwera „Srv2” nie może rozpocząć transakcji rozproszonej. (TRANSLATION: The operation could not be performed because the OLE DB server 'SQLNCLI11' for linked server 'Srv2' could not perform a servicing transaction.)
> 	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.executeStatement(JDBCPreparedStatementImpl.java:208)
> 	at org.jkiss.dbeaver.model.sql.exec.SQLScriptProcessor.executeStatement(SQLScriptProcessor.java:229)
> 	at org.jkiss.dbeaver.model.sql.exec.SQLScriptProcessor.lambda$0(SQLScriptProcessor.java:196)
> 	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:172)
> 	at org.jkiss.dbeaver.model.sql.exec.SQLScriptProcessor.executeSingleQuery(SQLScriptProcessor.java:193)
> 	at org.jkiss.dbeaver.model.sql.exec.SQLScriptProcessor.runScript(SQLScriptProcessor.java:119)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.processScript(SQLScriptExecuteHandler.java:156)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.runScripts(SQLScriptExecuteHandler.java:129)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.lambda$0(SQLScriptExecuteHandler.java:79)
> 	at org.jkiss.dbeaver.registry.task.TaskRunJob.run(TaskRunJob.java:133)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.executeWithSettings(SQLScriptExecuteHandler.java:77)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.executeTask(SQLScriptExecuteHandler.java:66)
> 	at org.jkiss.dbeaver.registry.task.TaskRunJob.executeTask(TaskRunJob.java:128)
> 	at org.jkiss.dbeaver.registry.task.TaskRunJob.run(TaskRunJob.java:95)
> 	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
> 	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
> Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Nie można wykonać operacji, ponieważ dostawca OLE DB „SQLNCLI11” dla połączonego serwera „Srv2” nie może rozpocząć transakcji rozproszonej. (TRANSLATION: The operation could not be performed because the OLE DB server 'SQLNCLI11' for linked server 'Srv2' could not perform a servicing transaction.)
> 	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:262)
> 	at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1632)
> 	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.doExecutePreparedStatement(SQLServerPreparedStatement.java:602)
> 	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement$PrepStmtExecCmd.doExecute(SQLServerPreparedStatement.java:524)
> 	at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7418)
> 	at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:3274)
> 	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:247)
> 	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:222)
> 	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.execute(SQLServerPreparedStatement.java:505)
> 	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.execute(JDBCPreparedStatementImpl.java:261)
> 	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.executeStatement(JDBCPreparedStatementImpl.java:205)
> 	... 15 more
> 2022-11-16 15:04:03.375 - 	Set JDBC connection auto-commit true
> 2022-11-16 15:04:03.376 - org.jkiss.dbeaver.DBException: Error executing script '/General/Scripts/TASKS/Script.sql'
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.runScripts(SQLScriptExecuteHandler.java:137)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.lambda$0(SQLScriptExecuteHandler.java:79)
> 	at org.jkiss.dbeaver.registry.task.TaskRunJob.run(TaskRunJob.java:133)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.executeWithSettings(SQLScriptExecuteHandler.java:77)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.executeTask(SQLScriptExecuteHandler.java:66)
> 	at org.jkiss.dbeaver.registry.task.TaskRunJob.executeTask(TaskRunJob.java:128)
> 	at org.jkiss.dbeaver.registry.task.TaskRunJob.run(TaskRunJob.java:95)
> 	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
> 	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
> Caused by: org.jkiss.dbeaver.model.exec.DBCException: Script execute failed
> 	at org.jkiss.dbeaver.model.sql.exec.SQLScriptProcessor.runScript(SQLScriptProcessor.java:168)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.processScript(SQLScriptExecuteHandler.java:156)
> 	at org.jkiss.dbeaver.tools.sql.task.SQLScriptExecuteHandler.runScripts(SQLScriptExecuteHandler.java:129)
> 	... 8 more
> Caused by: org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [7391] [S0002]: Nie można wykonać operacji, ponieważ dostawca OLE DB „SQLNCLI11” dla połączonego serwera „Srv2” nie może rozpocząć transakcji rozproszonej. (TRANSLATION: The operation could not be performed because the OLE DB server 'SQLNCLI11' for linked server 'Srv2' could not perform a servicing transaction.)
> 	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.executeStatement(JDBCPreparedStatementImpl.java:208)
> 	at org.jkiss.dbeaver.model.sql.exec.SQLScriptProcessor.executeStatement(SQLScriptProcessor.java:229)
> 	at org.jkiss.dbeaver.model.sql.exec.SQLScriptProcessor.lambda$0(SQLScriptProcessor.java:196)
> 	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:172)
> 	at org.jkiss.dbeaver.model.sql.exec.SQLScriptProcessor.executeSingleQuery(SQLScriptProcessor.java:193)
> 	at org.jkiss.dbeaver.model.sql.exec.SQLScriptProcessor.runScript(SQLScriptProcessor.java:119)
> 	... 10 more
> Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Nie można wykonać operacji, ponieważ dostawca OLE DB „SQLNCLI11” dla połączonego serwera „Srv2” nie może rozpocząć transakcji rozproszonej. (TRANSLATION: The operation could not be performed because the OLE DB server 'SQLNCLI11' for linked server 'Srv2' could not perform a servicing transaction.)
> 	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:262)
> 	at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1632)
> 	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.doExecutePreparedStatement(SQLServerPreparedStatement.java:602)
> 	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement$PrepStmtExecCmd.doExecute(SQLServerPreparedStatement.java:524)
> 	at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7418)
> 	at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:3274)
> 	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:247)
> 	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:222)
> 	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.execute(SQLServerPreparedStatement.java:505)
> 	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.execute(JDBCPreparedStatementImpl.java:261)
> 	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.executeStatement(JDBCPreparedStatementImpl.java:205)
> 	... 15 more
> 2022-11-16 15:04:03.377 - SQL script execute completed

Could you please help me to resolve this issue.
As I said above - If i run the same command in SQL Script Window - it works fine and done it's job. But Task Run, fails.

I also save that command in a procedure - but it didn't resolve my problem

Regards 
Kamil

**Comment by LonwoLonwo** [comment]
```
2022-11-16 15:04:03.261 - > Default catalog: XXX
2022-11-16 15:04:03.261 - > Default schema: xxx
```

Is this info correct for your query? Did you choose the right catalog for your task?

**Reply by albercik777** [reply]
Original messege looks fine. Catalog and schema are the same as my default script settings.
